conductor_api/src/conductor/passphrase_manager.rs

```diff
@@ -125,3 +127,80 @@ impl PassphraseService for PassphraseServiceMock {
         Ok(SecBuf::with_insecure_from_string(self.passphrase.clone()))
     }
 }
+
+pub struct PassphraseServiceUnixSocket {
+    path: String,
+    stream: Arc<Mutex<Option<std::io::Result<BufReader<UnixStream>>>>>,
+}
+
+impl PassphraseServiceUnixSocket {
+    pub fn new(path: String) -> Self {
+        let stream = Arc::new(Mutex::new(None));
+        let stream_clone = stream.clone();
+        let listener = UnixListener::bind(path.clone())
+            .expect("Could not create unix socket for passphrase service");
+        log_info!("Start accepting passphrase IPC connections on socket...");
+        thread::spawn(move || {
+            let accept_result = listener.accept();
+            {
+                *(stream_clone.lock().unwrap()) =
+                    Some(accept_result.map(|(stream, _)| BufReader::new(stream)));
+            }
+            log_info!("Passphrase provider connected through unix socket");
+        });
+        PassphraseServiceUnixSocket { path, stream }
+    }
+}
+
+impl Drop for PassphraseServiceUnixSocket {
+    fn drop(&mut self) {
+        std::fs::remove_file(self.path.clone()).unwrap();
+    }
+}
+
+impl PassphraseService for PassphraseServiceUnixSocket {
+    fn request_passphrase(&self) -> Result<SecBuf, HolochainError> {
+        log_debug!("Passphrase needed. Using unix socket passphrase service...");
+        while self.stream.lock().unwrap().is_none() {
+            log_debug!("No one connected via socket yet. Waiting...");
+            thread::sleep(Duration::from_millis(500));
+        }
+
+        log_debug!("We have an open connection to a passphrase provider.");
+
+        // Request and read passphrase from socket
+        let mut passphrase_string = {
+            let mut stream_option = self.stream.lock().unwrap();
```

Comment:
A lot of unwraps and expects in this block. Since the containing function returns a Result it might be better to early return the errors

Reply:
I've replaced all `unwrap`s with `expect`s which should give enough context if triggered. In this case I regard expects to be the right choice since the conductor can't start and do its job if there is no way to get a passphrase from the user, and this way we have the full stack trace when that happens, which we would not if we return an error that then lets the process panic somewhere else...